src/v/cloud_storage/partition_probe.h

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2022 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "model/fundamental.h"
+
+#include <seastar/core/metrics_registration.hh>
+
+namespace cloud_storage {
+
+class partition_probe {
+public:
+    partition_probe(const model::ntp& ntp);
+
+    void add_bytes_read(uint64_t read) { _bytes_read += read; }
+    void add_records_read(uint64_t read) { _records_read += read; }
+
+    void segment_added() { ++_cur_segments; }
+    void segment_materialized() { ++_cur_materialized_segments; }
+    void segment_offloaded() { --_cur_materialized_segments; }
+
+    void reader_created() { ++_cur_readers; }
+    void reader_destroyed() { --_cur_readers; }
+    void segment_reader_created() { ++_cur_segment_readers; }
+    void segment_reader_destroyed() { --_cur_segment_readers; }
+
+private:
+    uint64_t _bytes_read = 0;
+    uint64_t _records_read = 0;
+
+    int64_t _cur_segments = 0;
```

Comment:
nit: bytes+records make sense for 8 bytes, but for the next 4 fields you can save 16 bytes by making them in32_t as 2BN active readers would be probably world shattering record for a single partition.

Reply:
good catch

Reply:
good idea, fixed